internal/isoutil/isoutil.go

```diff
@@ -228,3 +236,69 @@ func (h *installerHandler) VolumeIdentifier() (string, error) {
 
 	return strings.TrimSpace(string(volumeId)), nil
 }
+
+func GetFileLocation(filePath, isoPath string) (int64, error) {
+	d, err := diskfs.OpenWithMode(isoPath, diskfs.ReadOnly)
+	if err != nil {
+		return 0, err
+	}
+
+	fs, err := d.GetFilesystem(0)
+	if err != nil {
+		return 0, err
+	}
+
+	isoFile, err := fs.OpenFile(filePath, os.O_RDONLY)
+	if err != nil {
+		return 0, errors.Wrapf(err, "Failed to open file %s", filePath)
+	}
+
+	myFile := isoFile.(*iso9660.File)
+	defaultSectorSize := int64(2 * 1024)
+	offset := int64(myFile.Location()) * defaultSectorSize
+	return offset, nil
+}
+
+func WriteHeader(ignitionOffsetInfo, ramDiskOffsetInfo *OffsetInfo, isoPath string) error {
+	iso, err := os.OpenFile(isoPath, os.O_WRONLY, 0o664)
+	if err != nil {
+		return err
+	}
+	defer iso.Close()
+
+	headerOffset := int64(32768) // first 32KB in ISO
+
+	// Write ignition config
+	headerOffset, err = writeOffsetInfo(headerOffset, ignitionOffsetInfo, iso)
+	if err != nil {
+		return err
+	}
+
+	// Write ram disk
+	_, err = writeOffsetInfo(headerOffset, ramDiskOffsetInfo, iso)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func writeOffsetInfo(headerOffset int64, offsetInfo *OffsetInfo, iso *os.File) (int64, error) {
+	buf := new(bytes.Buffer)
+	err := binary.Write(buf, binary.LittleEndian, offsetInfo)
```

Comment:
I think `binary.LittleEndian` should be a global const at the top of the file, in case we want to encode more fields - we would probably want endianness to be consistent (i.e. it's not specific just to this field)

Reply:
I'm actually not sure we want it as constant in isoutil, this function is an helper and not really part installerHandler, so we might want the extract it later on. I think we can keep it as an implementation detail for now.

Reply:
I didn't mean for it to be public - I just don't want it to be hardcoded in this specific line of code - because it feels like it's implying that this endianness is specific to this field, while the true intention is for it to the same endianness for all fields we're going to encode (for now, this is the only field). But say you wanted to add something like a version number (see comment below), it would also need an endianness - and that endianness should be a constant global to this file.